hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -90,8 +91,8 @@ data HydraMessage tx
 -- fully recoverable from transactions observed on chain
 data OnChainTx tx
   = InitTx (Set ParticipationToken)
-  | CommitTx ParticipationToken Natural
-  | CollectComTx
+  | CommitTx ParticipationToken (UTxO tx)
+  | CollectComTx (UTxO tx)
   | CloseTx (Snapshot tx) [tx]
   | ContestTx
   | FanoutTx (UTxO tx)
```

Comment:
Yeah.. this really now shows what we would like to have in on-chain-txs in an ideal world. Good starting point for today's engineering meeting I'd say!